ui/main/src/app/views/customCardList/CustomCardListView.ts

```diff
@@ -92,7 +92,16 @@ export class CustomCardListView {
     }
 
     public getBusinessPeriod(): {startDate: number; endDate: number} {
-        return RealTimeDomainService.getCurrentDomain();
+        if (this.customScreenDefinition.initialBusinessPeriod === 'FROM_TODAY_TO_YEAR_END') {
+            const now = new Date();
+            const currentYear = now.getFullYear();
+            return {
+                startDate: new Date(currentYear, now.getMonth(), now.getDate()).getTime(),
+                endDate: new Date(currentYear, 11, 31, 23, 59, 59, 999).getTime()
+            };
+        } else {
+            return RealTimeDomainService.getCurrentDomain();
```

Comment:
When you do that , you get the business period from today to year end but you do not set the Realtime Domain which means you have the good period on the screen but the domain is stil the one from the feed . 

So you need to call `RealTimeDomainService.setStartAndEndPeriod` method with the computed period FROM_TODAY_TO_YEAR_END


But doing that inside the current get method is not good practice (get method should not modify internal state) 

So i propose to set the domain in the constructor of the CustomCardListView when we have FROM_TODAY_TO_YEAR_END 

And when we will call getBusinessPeriod , we just have to call ` RealTimeDomainService.getCurrentDomain();`

Reply:
Done.